### ISSUE TITLE
plugins.youtube: 400 Client Error: Bad request for url: https://consent.youtube.com/s

### ISSUE THREAD

**Issue by hiboudev** [streamlink/streamlink#4513]
### Checklist

- [X] This is a plugin issue and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed plugin issues](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22plugin+issue%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

Latest stable release

### Description

I randomly get this error on Windows 10 when fetching YouTube URL using channel id :

> Unable to open URL: https://consent.youtube.com/s (400 Client Error: Bad Request for url: https://consent.youtube.com/s)

This is random, sometimes stream fetching works, sometimes no.

And this is related with using the following type of URL, using the channel id instead of a video id : 
https://www.youtube.com/c/TinyKittens/live
https://www.youtube.com/channel/UCeL2LSl91k2VccR7XEh5IKg/live

It always works fine with URLs containing the "watch" parameter : 
https://www.youtube.com/watch?v=Pss7fUAUqrs

I'm doing the exact same test on Debian with Python 3.7 and it always works fine.

### Debug log

```text
===== Output when it fails :

> streamlink https://www.youtube.com/c/TinyKittens/live --loglevel debug
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.10.1
[cli][debug] Streamlink: 4.0.1
[cli][debug] Requests(2.27.1), Socks(1.7.1), Websocket(1.3.2)
[cli][debug] Arguments:
[cli][debug]  url=https://www.youtube.com/c/TinyKittens/live
[cli][debug]  --loglevel=debug
[cli][debug]  --ffmpeg-ffmpeg=C:\Program Files (x86)\Streamlink\ffmpeg\ffmpeg.exe
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/c/TinyKittens/live
[plugins.youtube][debug] c_data_keys: gl, m, pc, continue, x, v, bl, hl, src, uxe, set_eom, set_ytc, set_apyt
error: Unable to open URL: https://consent.youtube.com/s (400 Client Error: Bad Request for url: https://consent.youtube.com/s)

===== Output when it randomly works :

> streamlink https://www.youtube.com/c/TinyKittens/live  --loglevel debug
[cli][debug] OS:         Windows 10
[cli][debug] Python:     3.9.10
[cli][debug] Streamlink: 3.2.0
[cli][debug] Requests(2.27.1), Socks(1.7.1), Websocket(1.2.3)
[cli][debug] Arguments:
[cli][debug]  url=https://www.youtube.com/c/TinyKittens/live
[cli][debug]  --loglevel=debug
[cli][debug]  --ffmpeg-ffmpeg=C:\Program Files (x86)\Streamlink\ffmpeg\ffmpeg.exe
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/c/TinyKittens/live
[plugins.youtube][debug] c_data_keys: gl, m, pc, continue, ca, x, v, t, hl, src, uxe
[plugins.youtube][debug] Using video ID: Pss7fUAUqrs
[plugins.youtube][debug] This video is live.
[utils.l10n][debug] Language code: fr_FR
Available streams: 144p (worst), 240p, 360p, 480p, 720p (best)
```

**Comment by bastimeyer** [comment]
Had a quick look at the issue, and from what it looks like, YouTube is showing two different consent dialogs now:

1. one with "I accept" / "Customize" buttons and a single form with the target `https://consent.youtube.com/s` (the one which the plugin is built for)
2. one with "Accept all" / "Decline all" / "More options" buttons and four forms with the target `https://consent.youtube.com/save`

I have no idea why both dialogs pop up randomly instead of just one.

The problem with the second dialog is that I don't know which form to use for the consent data. It seems like the two forms for the "Accept all" buttons have the following input fields set whereas the other ones do not:

```html
<input type="hidden" name="set_ytc" value="true" required="">
<input type="hidden" name="set_apyt" value="true" required="">
```

The plugin's consent validation schema and its xpath queries need to be adjusted for both consent dialogs. Let me see...

**Comment by campones** [comment]
it's pretty weird, I was reading this threat yesterday as I met the same issue. Then I updated my streamlink version to the last release, and it worked out perfectly out of the box

today.. I get the consent 400 bad request again? I don't get it 

`/usr/local/bin/streamlink -l debug https://www.youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g/live 480p
[cli][info] streamlink is running as root! Be careful!
[cli][debug] OS:         Linux-5.13.0-44-generic-x86_64-with-glibc2.29
[cli][debug] Python:     3.8.10
[cli][debug] Streamlink: 4.0.1
[cli][debug] Requests(2.27.1), Socks(1.7.1), Websocket(1.3.2)
[cli][debug] Arguments:
[cli][debug]  url=https://www.youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g/live
[cli][debug]  stream=['480p']
[cli][debug]  --loglevel=debug
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g/live
[plugins.youtube][debug] c_data_keys: gl, m, pc, continue, x, v, bl, hl, src, uxe, set_eom, set_ytc, set_apyt
error: Unable to open URL: https://consent.youtube.com/s (400 Client Error: Bad Request for url: https://consent.youtube.com/s)`

**Comment by bastimeyer** [comment]
> [cli][debug] Streamlink: 4.0.1
> [plugins.youtube][debug] c_data_keys: gl, m, pc, continue, x, v, bl, hl, src, uxe, set_eom, set_ytc, set_apyt

You can see in the PR which closed the issue that the "c_data_keys" log line was replaced. You are using a modified version of 4.0.1.

https://github.com/streamlink/streamlink/pull/4515/files#diff-9a50f4669ceb763d43870dd8857c0c376dd0596f75199fa8b53c3778c42f1bf4L260

```
$ streamlink -l debug youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g/live 480p
[cli][debug] OS:         Linux-5.17.8-1-git-x86_64-with-glibc2.35
[cli][debug] Python:     3.10.4
[cli][debug] Streamlink: 4.0.1+50.g588cb1da
[cli][debug] Requests(2.27.1), Socks(1.7.1), Websocket(1.3.2)
[cli][debug] Arguments:
[cli][debug]  url=youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g/live
[cli][debug]  stream=['480p']
[cli][debug]  --loglevel=debug
[cli][debug]  --player=mpv
[cli][info] Found matching plugin youtube for URL youtube.com/channel/UCCCPCZNChQdGa9EkATeye4g/live
[plugins.youtube][debug] consent target: https://consent.youtube.com/save
[plugins.youtube][debug] consent data: gl, m, pc, continue, x, v, bl, hl, src, uxe, set_ytc, set_apyt, set_eom
[plugins.youtube][debug] Using video ID: u9foWyMSATM
[plugins.youtube][debug] This video is live.
[utils.l10n][debug] Language code: en_US
[cli][info] Available streams: 144p (worst), 240p, 360p, 480p, 720p, 1080p (best)
[cli][info] Opening stream: 480p (hls)
...
```

**Comment by campones** [comment]
Might be. I am updating as usual
Pip install --upgrade streamline

And as I said yesterday last night right after updating was working fine

**Comment by bastimeyer** [comment]
> /usr/local/bin/streamlink
> [cli][info] streamlink is running as root! Be careful!

> Might be. I am updating as usual

Whatever you're trying to do, you are doing it wrong. This very much looks like you're mixing python environments or you're doing other kinds of modifications. You also commented here lots of times in the past with several python and/or outdated streamlink installs, also with custom ffmpeg builds that weren't working. I have clearly shown you that 4.0.1 and the old log line are incompatible, and also that a clean install of 4.0.1 is working fine.
https://streamlink.github.io/install.html

If you can prove that the latest version is indeed broken, then open a new issue. This one has been resolved by #4515.

**Comment by bastimeyer** [comment]
I updated my comment because I realized that #4515 wasn't part of 4.0.1. Doesn't matter though, as the issue is fixed on the master branch.
